docs: replace occurrences of "charmcraft.yaml" and "recipe" (#2154)
canonical/charmcraft: docs/tutorial/flask.rst

    :end-before: [docs:create-charm-dir-end]
    :dedent: 2

> We'll need a project file named ``charmcraft.yaml``, ``requirements.txt`` and source

Reviewer: ```suggestion
We'll need the project's ``charmcraft.yaml``, ``requirements.txt`` and source
```

Author: https://github.com/canonical/charmcraft/pull/2154/commits/6e57a8e7fa62014b0fdf6fd03578db84ab8172a0

    :dedent: 2

Change back into the charm directory using ``cd charm``. The ``flask-framework``

> Charmcraft extension supports adding configurations to the project which will be passed

Reviewer: Extensions act as a preprocessor steps for project files.

```suggestion
Charmcraft extension supports adding configurations to the project file which will be passed
```

Author: https://github.com/canonical/charmcraft/pull/2154/commits/6e57a8e7fa62014b0fdf6fd03578db84ab8172a0
